modules/drivers/mongo/test/metabase/driver/mongo/connection_test.clj

```diff
@@ -0,0 +1,143 @@
+(ns metabase.driver.mongo.connection-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.driver.mongo.connection :as mongo.connection]
+   [metabase.driver.mongo.database :as mongo.db]
+   [metabase.driver.mongo.util :as mongo.util]
+   [metabase.driver.util :as driver.u]
+   [metabase.test :as mt])
+  (:import
+   (com.mongodb ServerAddress)
+   (com.mongodb.client MongoDatabase)))
+
+;;;; TODO: move some tests to db-test?
+
+(set! *warn-on-reflection* true)
+
+(def ^:private mock-details
+  {:user "test-user"
+   :pass "test-pass"
+   :host "test-host.place.com"
+   :dbname "datadb"
+   :authdb "authdb"
+   :use-srv true})
+
+(deftest ^:parallel fqdn?-test
+  (testing "test hostname is fqdn"
+    (is (true? (#'mongo.db/fqdn? "db.mongo.com")))
+    (is (true? (#'mongo.db/fqdn? "replica-01.db.mongo.com")))
+    (is (false? (#'mongo.db/fqdn? "localhost")))
+    (is (false? (#'mongo.db/fqdn? "localhost.localdomain")))))
+
+;; TODO: remove this!
+(defn- details-normalized [details]
+  (#'mongo.db/details-normalized details))
+
+(deftest ^:parallel srv-conn-str-test
+  (let [db-details {:user "test-user"
+                    :pass "test-pass"
+                    :host "test-host.place.com"
+                    :dbname "datadb"
+                    :authdb "authdb"
+                    :use-srv true}]
+    (testing "mongo+srv connection string used when :use-srv is thruthy"
+      (is (re-find #"\Qmongodb+srv://test-user:test-pass@test-host.place.com/datadb?authSource=authdb\E"
```

Comment:
Isn't this what we usually write as `str/includes?`?

Reply:
`re-find` is definitely not necessary here. Thanks.